app/javascript/components/users/SignInComponent.tsx

```diff
@@ -57,47 +69,153 @@ function SignInComponent(props:SignInComponentProps) : JSX.Element {
-
-					{componentState === 'submitting' ? "" : <>
-						<div data-testid="signInErrorDiv">{ failed ? lastError.data.error.map((i) => i).join('; ') : ""}</div>
-						<div data-testid="currentUserDiv">{currentUser ? currentUser.id : ""}</div>
-					</>
-					}
-
-				</Form>;
+					setIsValid(isValid);
+				}, [isValid]);
+
+				//Form
+				return (
+					<Form>
+						{/* NOTE: if a Button should submit a form, mark it as type="submit". Otherwise pressing Enter won't submit form*/}
+									<Box display="flex" justifyContent="center" alignItems="center">
+										<Box p={1.5}>
+											<Field component={TextField} name="email" type="text"
+												label={emailLabel}
+												InputProps={{
+													startAdornment: (
+														<InputAdornment position="start">
+															<AccountCircle fontSize="small" />
+														</InputAdornment>
+													),
+												}}
+											/>
+										</Box>
+									</Box>
+									<Box display="flex" justifyContent="center" alignItems="center">
+										<Box p={1.5}>
+											<Field component={TextField} name="password" type="password"
+												label={passwordLabel}
+												InputProps={{
+													startAdornment: (
+														<InputAdornment position="start">
+															<LockOpenIcon fontSize="small" />
+														</InputAdornment>
+													),
+												}} />
+										</Box>
+									</Box>
+                    <Box display="flex" justifyContent="center" alignItems="center">
+                      {componentState === 'submitting' ? "" : <>
+							          {failed ? <p>{emailValidLabel}</p> : ""}
+                        </>
+                      }
+                    </Box>
+                    <Box display="flex" justifyContent="center" alignItems="center">
+                      {componentState === 'canSubmit' && !touched.email && !touched.password ?
+                        <p>{enterInfoLabel}</p>
+                        : null
+                      }
+                    </Box>
+									<Box p={2} display="flex" justifyContent="center" alignItems="center">
+                    {componentState !== 'submitting' && isValid  && !failed && touched.email ? 
+                      <Button
+                        data-testid="signInButton"
+                        type="submit"
+                        color="primary"
+                        variant='contained'
+                      > 
+                        {formatMessage({ id: 'submit' })} 
+                      </Button>
+                    : null }
+                    {componentState !== 'submitting' && failed && isValid ?
+                      <Button
+                        data-testid="signInButton"
+                        type="submit"
+                        color="primary"
+                        variant='contained'
+                      >
+                        {formatMessage({ id: 'submit' })}
+                      </Button>
```

Comment:
So, I'm not sure on the decision here. I think the sign in button should be visible all the time EXCEPT during submitting or success but it should be disabled when you can't submit. To do this, set the disabled prop on the button. https://material-ui.com/api/button/

Reply:
We've changed this.